src/py/flwr/client/app.py

```diff
@@ -146,20 +178,31 @@ class `flwr.client.Client` (default: None)
     """
     event(EventType.START_CLIENT_ENTER)
 
-    _check_actionable_client(client, client_fn)
+    if load_callable_fn is None:
+        _check_actionable_client(client, client_fn)
 
-    if client_fn is None:
-        # Wrap `Client` instance in `client_fn`
-        def single_client_factory(
-            cid: str,  # pylint: disable=unused-argument
-        ) -> Client:
-            if client is None:  # Added this to keep mypy happy
-                raise Exception(
-                    "Both `client_fn` and `client` are `None`, but one is required"
-                )
-            return client  # Always return the same instance
+        if client_fn is None:
+            # Wrap `Client` instance in `client_fn`
+            def single_client_factory(
+                cid: str,  # pylint: disable=unused-argument
+            ) -> Client:
+                if client is None:  # Added this to keep mypy happy
+                    raise Exception(
+                        "Both `client_fn` and `client` are `None`, but one is required"
+                    )
+                return client  # Always return the same instance
+
+            client_fn = single_client_factory
+
+        def _load_app() -> Flower:
+            return Flower(client_fn=client_fn)
 
-        client_fn = single_client_factory
+        load_callable_fn = _load_app
+    else:
+        warn_experimental_feature("`load_callable_fn`")
+
+    # At this point, only `load_callable_fn` should be used
+    # Both `client` and `client_fn` must not be used directly
 
```

Comment:
Are we planning to deprecate `client_fn`? Maybe you can explain the thinking in the comment.

Reply:
This is unrelated to deprecation. The comment merely indicates that, after this line, `client` and `client_fn` should be considered out of scope. Not using those two directly, but only through `load_callable_fn`, ensures that all three ways of starting a client behave (almost) exactly the same. If also frees us from having if/else statements in many places because we handle all cases via `load_callable_fn`.

Reply:
Thx for explaining!